apps/language_server/lib/language_server/server.ex

```diff
@@ -496,11 +496,15 @@ defmodule ElixirLS.LanguageServer.Server do
       |> Enum.map(&String.to_atom/1)
 
     if dialyzer_enabled?(state),
-      do: Dialyzer.analyze(state.build_ref, warn_opts, state.settings["dialyzerFormat"])
+      do: Dialyzer.analyze(state.build_ref, warn_opts, dialyzer_default_format(state))
 
     state
   end
 
+  defp dialyzer_default_format(state) do
+    state.settings["dialyzerFormat"] || "dialyxir_long"
```

Comment:
Should be just `"dialyxir".

https://github.com/jeremyjh/dialyxir#command-line-options`

Reply:
Based on how the code works right now "dialyxir_long" is needed. Also since that's the value used in the end-user config file we'll probably want to keep it as-is. For reference here is how the format setting is used: https://github.com/elixir-lsp/elixir-ls/blob/8f6097dc12b9b3eaeee1f17788eac4f996f25953/apps/language_server/lib/language_server/dialyzer.ex#L435-L443

Which could probably be refactored as well (since it's basically calling into Dialyxir internals), but that would probably be best in a different PR.